source/slang/check.cpp

```diff
@@ -2597,11 +2697,35 @@ namespace Slang
                 UInt paramIndex = paramCounter++;
                 if( paramIndex < argCount )
                 {
-                    auto arg = attr->args[paramIndex];
+                    auto & arg = attr->args[paramIndex];
 
                     // TODO: support checking the argument against the declared
                     // type for the parameter.
-
+                    bool typeChecked = false;
+                    if (!paramDecl->getType())
+                        continue;
+                    if (auto basicType = paramDecl->getType()->AsBasicType())
+                    {
+                        if (basicType->baseType == BaseType::Int)
+                        {
+                            if (auto cint = checkConstantIntVal(arg))
+                            {
+                                attr->intArgVals[(uint32_t)paramIndex] = cint;
```

Comment:
The current `TryConstantFoldExpr` function returns an `IntVal` for those that can be folded (and only `int` type is implemented). That means the results can't just be appended to the back of the args array (since `Val` is not `Expr`). For now this is just a workaround.